enhancements/cloud-integration/tokenized-auth-enablement-operators-on-cloud.md

```diff
@@ -0,0 +1,404 @@
+CredentialsRequest objects. This is a new API field on the CredentialsRequest spec. When CCO acquires a 
+CredentialsRequest it will create a Secret (as today mostly) with new functionality: Adding to the Secret a path to the
+projected ServiceAccount token (static for OCP) and a roleARN (from the new field on the CredentialsRequest).
+
+CCO with the changes anticipated here, operating in Manual mode, will continue to ignore CredentialsRequests that do not
+contain the new STS related fields and as these are new fields, existing clusters won't have CredentialsRequests with 
+those fields already.
+
+Validation of CredentialsRequest by this new token-aware CCO? Maybe. Currently, CCO gets some permissions under ROSA
+(unused), could expand these permissions to include tools like AWS's Policy Simulator such that it could validate a role
+has permissions, and with this CCO could be the alerting mechanism for a changed CredentialsRequest without sufficient 
+permissions.
+
+CCO should consider adding the above, three-part test as a utility function exposed in the API.
+
+**HyperShift changes**: Include Cloud Credential Operator with token-aware mode (see above). Allows for processing of 
+CredentialsRequest objects added by operators.
+
+**OperatorHub and Console changes**: Allow for input from user of additional fields during install depending on the 
+Cloud Provider which will result in ENV variables on the Subscription object that are required in the 
+CredentialsRequests created by the Operator. Setting the Subscription config.ENV will allow UX for the information 
+needed by CCO or webhook, for input of the cloud credentials while not having to change the Subscription API.
+CLI Users will need to add the following to their subscription:
+
+```yaml
+kind: Subscription
+metadata:
+ name: ...
+spec:
+  config:
+    env:
+    - name: ROLEARN
+      value: "<role ARN >"
+```
+
+Show in OperatorHub that the cluster is in a mode that supports token-based authentication by reading the 
+`.spec.serviceAccountIssuer` from the Authentication CR, `.status.platformStatus.type` from the Infrastructure CR,
+`.spec.credentialsMode` from the CloudCredentials CR.
+
+Show that the operator is enabled for Token-based use by reading the CSV annotation provided by the operator author.
+
+Generate a manual ack for operator upgrade when cloud resources are involved. This can be determined by parsing the 
+Subscription objects. Ensures admin signs off that cloud resources are provisioned and working as intended before an 
+operator upgrade. This could be by setting the default subscription mode to Manual in the UI for operators that support 
+TAT.
+
+**Operator team changes**: Follow new guidelines for allowing for the operator to work on token auth 
+enabled cluster. New guidelines would include the following to use CCO has detected cluster is using time-based tokens:
+
+- Changing CSV to add CredentialRequest RBAC
```

Comment:
I don't recall seeing CSV defined before, might be useful for those with less context to spell it out on the first usage (eg. `Machine Config Operator (MCO)`)

Reply:
https://olm.operatorframework.io/docs/concepts/crds/clusterserviceversion/

but yes, spelling it out is a good idea since not everyone is steeped in OLM-land

Reply:
That link is also really helpful for those uninitiated in OLM (me!), thanks

Reply:
adding link to define it. ty.